drivers/SmartThings/matter-thermostat/src/init.lua

```diff
@@ -686,6 +692,30 @@ local function temp_event_handler(attribute)
   end
 end
 
+local temp_attr_handler_factory = function(minOrMax)
+  return function(driver, device, ib, response)
+    if ib.data.value == nil then
+      return
+    end
+    local temp = ib.data.value / 100.0
+    local unit = "C"
+    set_field_for_endpoint(device, minOrMax, ib.endpoint_id, temp)
+    local min = get_field_for_endpoint(device, setpoint_limit_device_field.MIN_TEMP, ib.endpoint_id)
+    local max = get_field_for_endpoint(device, setpoint_limit_device_field.MAX_TEMP, ib.endpoint_id)
+    -- Only emit the capability for RPC version >= 5 (unit conversion for
+    -- temperature range capability is only supported for RPC >= 5)
+    if version.rpc >= 5 and min ~= nil and max ~= nil then
+      if min < max then
+        device:emit_event_for_endpoint(ib.endpoint_id, capabilities.temperatureMeasurement.temperatureRange({ value = { minimum = min, maximum = max }, unit = unit }))
+        set_field_for_endpoint(device, setpoint_limit_device_field.MIN_TEMP, ib.endpoint_id, nil)
+        set_field_for_endpoint(device, setpoint_limit_device_field.MAX_TEMP, ib.endpoint_id, nil)
+      else
```

Comment:
Same comment as above, I think for this else case, it would still be useful to print that log even if we don't emit the capability event. These values will still be used in `set_setpoint`, so it'd be good to know if they are valid, even for older hubs

Reply:
Changed as suggested in the latest commit!